src/languages/gcode.js

```diff
@@ -7,58 +7,116 @@
  */
 
 export default function(hljs) {
-  const GCODE_IDENT_RE = '[A-Z_][A-Z0-9_.]*';
-  const GCODE_CLOSE_RE = '%';
   const GCODE_KEYWORDS = {
-    $pattern: GCODE_IDENT_RE,
-    keyword: 'IF DO WHILE ENDWHILE CALL ENDIF SUB ENDSUB GOTO REPEAT ENDREPEAT '
-      + 'EQ LT GT NE GE LE OR XOR'
-  };
-  const GCODE_START = {
-    className: 'meta',
-    begin: '([O])([0-9]+)'
+    $pattern: /[A-Z]+|%/,
+    meta: [
```

Comment:
```suggestion
    keyword: [
```

The default keywords for any grammar should be scoped `keyword`.